src/ucp/core/ucp_mm.h

```diff
@@ -41,9 +41,25 @@ typedef struct ucp_mem {
  */
 typedef struct ucp_mem_desc {
     ucp_mem_h                     memh;
+    ucp_mem_h                     meta_memh;
+    void                          *ptr;
+    void                          *meta_ptr;
+    ucs_memory_type_t             mem_type;
+    volatile uint32_t             count;
 } ucp_mem_desc_t;
 
 
+/**
+ * Memory pool private data descriptor.
+ */
+typedef struct ucp_rndv_mpool_priv {
+    ucp_worker_h                  worker;
+    ucs_memory_type_t             mem_type;
+    int                           num_frags;
```

Comment:
why needed?

Reply:
We don't use the fragment size passed to mpool init in case of rndv mpool frags because that size only corresponds to metadata elem size. Instead we manually populate num_frags and frag_size to request a specific payload size.

Reply:
pls convert to unsigned